### PR TITLE
Sentieon bwa mem

### PR DESCRIPTION
**Pull request by asp8200**
This PR contains a new module for running sentieon bwa mem.

Four GitHub-secrets (SENTIEON_LICENSE_BASE64, SENTIEON_AUTH_MECH_BASE64, SENTIEON_ENCRYPTION_KEY and SENTIEON_LICENSE_MESSAGE) have been introduced in this repo in order to run the sentieon-tests using a test license.

Two of those gitHub-secrets (SENTIEON_ENCRYPTION_KEY and SENTIEON_LICENSE_MESSAGE) needs to be processed by `.github/workflows/pytest-workflow.yml` calling `tests/modules/nf-core/sentieon/license_message.py` thus creating a fifth variable needed for the test-licensing to work. (If you want that python-script placed elsewhere, just let me know.)

Added `modules/nf-core/sentieon/setup.md` which describe how to setup the sentieon license as a nextflow secret.

Using NGC's Sentieon-license, I also tested that the sentieon-bwamem-module works with a regular license passed into the module as a nextflow-secret `SENTIEON_LICENSE_BASE64`.

Finally, the sentieon-license-variables where removed from the sentieon-bwaindex-module as it turns out they are not needed for the particular sentieon-function.

- [x] This PR  contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md) (Yeah, pretty much I'd say. I didn't do the git rebase thing.)
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.  (No, Maxime built a dockercontainer for this, and put it on dockerhub) 
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by maxulysse** [comment]
Can you add the documentation you wrote as a MD file at the base of the sentieon folder?

**Comment by asp8200** [comment]
> Can you add the documentation you wrote as a MD file at the base of the sentieon folder?

Like here `modules/nf-core/sentieon` ?  What do you what the file to be called? `licensing.md`, `setup.md`, `readme.md`?

**Comment by maxulysse** [comment]
> > Can you add the documentation you wrote as a MD file at the base of the sentieon folder?
> 
> Like here `modules/nf-core/sentieon` ?  What do you what the file to be called? `licensing.md`, `setup.md`, `readme.md`?

Exactly

**Comment by nicorap** [comment]
Looks good ;). @FriederikeHanssen @maxulysse we need to have some reviews done I guess?

**Comment by maxulysse** [comment]
I'm on it

File: modules/nf-core/sentieon/bwamem/main.nf [hunk 1]
```diff
@@ -29,16 +29,17 @@ process SENTIEON_BWAMEM {
     def args = task.ext.args ?: ''
     def args2 = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def sentieon_auth_mech_base64 = task.ext.sentieon_auth_mech_base64 ?: ''
+    def sentieon_auth_data_base64 = task.ext.sentieon_auth_data_base64 ?: ''
```
Comment (maxulysse) [inline modules/nf-core/sentieon/bwamem/main.nf:33]: So this is only when using the nf-core license server?

**Review by maxulysse — APPROVED** [review]
Looks good to me

File: modules/nf-core/sentieon/bwamem/main.nf (hunk 1 above)
Comment (maxulysse) [reply modules/nf-core/sentieon/bwamem/main.nf:33]: yes, clearly stated as such in the comments

**Comment by maxulysse** [comment]
Just sync the branch to master, and now I see that in the failed tests:
> `Authentication failed`

**Comment by asp8200** [comment]
> Just sync the branch to master, and now I see that in the failed tests:
> 
> > `Authentication failed`

That is odd. The docker test is failing while the singularity test is passing.

**Review by maxulysse — APPROVED** [review]

**Pull request merged**